### DISCUSSION TITLE
v0.18.0

### DISCUSSION THREAD

**Discussion by fbs** [bpftrace/bpftrace#2597] [Development]
#### Added
- Add `iter:task_vma` iterators detection
  - [#2524](https://github.com/iovisor/bpftrace/pull/2524)
- Support parsing bitfields from BTF/DWARF
  - [#2505](https://github.com/iovisor/bpftrace/pull/2505)
- Support printing entire structs
  - [#2557](https://github.com/iovisor/bpftrace/pull/2557)
- BTF support for tracepoints defined in modules
  - [#2479](https://github.com/iovisor/bpftrace/pull/2479)
- Add trailer to truncated strings
  - [#2559](https://github.com/iovisor/bpftrace/pull/2559)
- Enable watchpoint support for PowerPC
  - [#2577](https://github.com/iovisor/bpftrace/pull/2577)
- Add new function, `offsetof`, get the offset of the element in the struct
  - [#2579](https://github.com/iovisor/bpftrace/pull/2579)
- Add 'StackMode::raw' for ustack and kstack formatting
  - [#2581](https://github.com/iovisor/bpftrace/pull/2581)
- Add 'BPFTRACE_STACK_MODE' env variable
  - [#2586](https://github.com/iovisor/bpftrace/pull/2586)

#### Changed
- Improve attaching to uprobes with size 0
  - [#2562](https://github.com/iovisor/bpftrace/pull/2562)

#### Fixed
- Simplify and fix probe index assignment
  - [#2482](https://github.com/iovisor/bpftrace/pull/2482)
- Handle colon in positional param used in attachpoint
  - [#2514](https://github.com/iovisor/bpftrace/pull/2514)
- Handle BPF load errors for multi-attachpoints
  - [#2521](https://github.com/iovisor/bpftrace/pull/2521)
- Respect BPFTRACE_STRLEN environment variable for all strings
  - [#2545](https://github.com/iovisor/bpftrace/pull/2545)
- Treat str() builtin's len parameter as int64
  - [#2546](https://github.com/iovisor/bpftrace/pull/2546)
- arm64: define the KASAN_SHADOW_SCALE_SHIFT macro
  - [#2518](https://github.com/iovisor/bpftrace/pull/2518)
- Fix segfaults in dwarf_parser
  - [#2587](https://github.com/iovisor/bpftrace/pull/2587)
 
#### Removed

<hr /><em>This discussion was created from the release <a href='https://github.com/iovisor/bpftrace/releases/tag/v0.18.0'>v0.18.0</a>.</em>

**Comment by aaronlu** [comment]
Will there be binary release for 0.18.0? Thanks.
I personally quite like the previous binary releases as it can run on any distro and doesn't depend on anything distro provides so it kind of always works.

**Reply by fbs** [reply]
yes sorry, ive been slacking. They're built by the CI after the release is made, I just have to pull them in :)

**Reply by fbs** [reply]
assets are now there, thanks for reminding me!

**Reply by aaronlu** [reply]
Great, thanks!
